sdk/servicebus/service-bus/src/core/messageReceiver.ts

```diff
@@ -392,13 +393,18 @@ export class MessageReceiver extends LinkEntity {
                 bMessage.messageId as string,
                 setTimeout(async () => {
                   try {
+                    let receiverName;
+                    if (this._receiver) {
+                      receiverName = this._receiver.name;
+                    }
```

Comment:
Can you do check on what  value does`this.name` at this point have?
I have a suspicion that it might be the same as `this._receiver.name`....

Reply:
I recall trying to use this, it was coming as undefined. Hence had made these changes to expose _receiver name

Reply:
The fact that auto lock renewal has started, it means that a message was received, meaning a receiver got created at some point in time.

Rhea receivers get created in the `init()` function.
`init()` function takes in `ReceiverOptions` which contains the name that gets used to create the rhea receiver.

If you follow the 2 ways `init()` gets called, once when this class gets instantiated, the other inside `detached()`, you will see that `this.name` and the name in the `options` sent to create the rhea receiver are the same.

Therefore, we can skip the `if` check here and the need for a new variable, and directly use `this.name` when calling `this._context.managementClient!.renewLock`

The main reason I am pushing for this is I rather reduce the number of variables that is tracking the name and have one source of truth instead

Reply:
Thanks for fixing this with https://github.com/Azure/azure-sdk-for-js/pull/2338/commits/3d6e5f5ae16491c72eadcf1e55948b98af244413